extensions/jupiter-aggregator/src/api.ts

```diff
@@ -0,0 +1,80 @@
+import { showToast, Toast, getPreferenceValues } from "@raycast/api";
+import { useFetch } from "@raycast/utils";
+import axios from "axios";
+
+interface Preferences {
+  TOKEN_LIST: string;
+}
+
+const preferences = getPreferenceValues<Preferences>();
+const list = preferences.TOKEN_LIST;
+
+export interface PriceResponse {
+  data: Sol;
+  timeTaken: number;
+  contextSlot: number;
+}
+
+interface Sol {
+  SOL: SolItem;
+}
+
+interface SolItem {
+  id: string;
+  mintSymbol: string;
+  vsToken: string;
+  vsTokenSymbol: string;
+  price: string;
+}
+
+export interface TokenListResponse {
+  address: string;
+  chainId: number;
+  decimals: number;
+  name: string;
+  symbol: string;
+  logoURI: string;
+  extensions: object;
+  tags: string;
+}
+
+export function fetchPrice(token) {
+  return useFetch<PriceResponse>(`https://price.jup.ag/v4/price?ids=${token}`, {
+    onError: (error: Error) => {
+      showToast({
+        style: Toast.Style.Failure,
+        title: "Price Fetch Failed",
+        message: "Check network connection",
+      });
+      console.log(error);
+    },
+  });
+}
+
+export function fetchTokenList() {
+  return useFetch<TokenListResponse>(`https://token.jup.ag/${list}`, {
+    onError: (error: Error) => {
+      showToast({
+        style: Toast.Style.Failure,
+        title: "Failed to fetch token list",
+        message: "Check network connection",
+      });
+      console.log(error);
+    },
+  });
+}
+
+export const compareTokens = async (token1: string, amount: any, token2: string) => {
+  try {
+    const res = await axios.get(`https://price.jup.ag/v4/price?ids=${token1}&vsToken=${token2}`);
```

Comment:
Did you consider using `useFetch`? https://developers.raycast.com/utilities/react-hooks/usefetch

Reply:
I was actually curious if you knew of a good way to do that. I'm already using fetchTokenList in the compare-token-prices command and I couldn't figure out how to implement two useFetch hooks in the same command. It would error out because there would be two sets of data:
```javascript
const { data, isLoading } = fetchTokenList()
const { data, isLoading } = compareTokens()
```
Any thoughts?

Reply:
Awesome!! Thank you!! I've started implementing it but got a little stuck. The main problem is preventing the data from executing before submitting the form / revalidating with the new inputs. I tried the `execute: false` in the fetch options and tried adding a `revalidate()` in the submit handler but it doesn't seem to be working. Any thoughts on how I could fix it? I submitted the changes so you can see what I'm doing

Reply:
Ok so I think I solved some problems by updating the interface for the result to target the right dataset. However now my problem is that I'm having trouble multiplying the data result with the amount. This was working before and I've tried a few things with no luck. Maybe I need to adjust my interface? I was using "string" for price before and it was working, but even if I parse them both as numbers still no luck :/ 

New Interface 
```javascript
export interface PriceResponse<T extends string> {
  data: Record<T, SolItem>;
  timeTaken: number;
  contextSlot: number;
}

interface SolItem {
  id: string;
  mintSymbol: string;
  vsToken: string;
  vsTokenSymbol: string;
  price: string;
}
```
Function trying to calculate 
```javascript
  const { data: tokenData, isLoading: tokenDataLoading, revalidate } = compareTokens(token1, token2);

  const submitHander = async (amount: any) => {
    revalidate()
      console.log(tokenData.data[token1].price)

      const total = Number(tokenData.data[token1].price) * Number(amount)
      console.log(total)

      push(<Result price={tokenData.data[token1].price} token2={token2} />);
};
```